core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/ExclusionFilters.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.container.analysis;
+
+import java.util.Arrays;
+import org.apache.commons.lang.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.PathPattern;
+import org.sonar.api.scan.filesystem.FileExclusions;
+
+public class ExclusionFilters {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExclusionFilters.class);
+
+  private final FileExclusions exclusionSettings;
+
+  private PathPattern[] mainInclusions;
+  private PathPattern[] mainExclusions;
+  private PathPattern[] testInclusions;
+  private PathPattern[] testExclusions;
+
+  public ExclusionFilters(ServerSettingsProvider serverSettingsProvider) {
+    this.exclusionSettings = new FileExclusions(serverSettingsProvider.getServerSettings());
+  }
+
+  public void prepare() {
+    mainInclusions = prepareMainInclusions();
+    mainExclusions = prepareMainExclusions();
+    testInclusions = prepareTestInclusions();
+    testExclusions = prepareTestExclusions();
+    log("Included sources: ", mainInclusions);
+    log("Excluded sources: ", mainExclusions);
+    log("Included tests: ", testInclusions);
+    log("Excluded tests: ", testExclusions);
+  }
+
+  private static void log(String title, PathPattern[] patterns) {
+    if (patterns.length > 0) {
+      LOG.debug(title);
+      for (PathPattern pattern : patterns) {
+        LOG.debug("  " + pattern);
+      }
+    }
+  }
+
+  public boolean accept(InputFile inputFile, InputFile.Type type) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "accept" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)